source/models/prediction.py

```diff
@@ -30,49 +30,43 @@ def value_check(feature_dict: dict) -> list:
         feature_dict (dict) dictionary of the features,
             derived from the cfg dictionary object.
     """
-    edited_input = []
-    while len(edited_input) != len(feature_dict):
-        for count, (key, value) in enumerate(feature_dict.items(), 1):
-            print(f"Write the data you want to be predicted {count}/4:")
+    features_input = []
+
+    for count, (key, value) in enumerate(feature_dict.items(), 1):
+        while len(features_input) != count:
```

Comment:
Now the loop works, but I find it a little bit weird to do it in this way.
A better/cleaner solution IMO is:
```python
for count, (key, value) in enumerate(feature_dict.items(), 1):
    while True:
        ...
        if value['min'] <= float(user_input) <= value['max']:
            features_input.append(user_input)
            break
        else:
            print(f"Please enter a Float that's between {value['min']} and {value['max']}")
```

Reply:
Very true! I didn't think of that way of doing it. I thought it was bad practice to do it like that but having a decisive break instead of a complicated condition to follow would be way faster execution wise.